src/sys/unix.rs

```diff
@@ -1315,6 +1317,31 @@ impl crate::Socket {
         .map(|_| ())
     }
 
+    /// Sets the value for `IP_BOUND_IF` option on this socket.
+    ///
+    /// If a socket is bound to an interface, only packets received from that
+    /// particular interface are processed by the socket.
+    ///
+    /// If `interface` is `None`, the binding is removed. If the `interface`
+    /// can't be resolved to a valid interface index, an error is returned.
```

Comment:
Well, I checked the man page for [`if_nametoindex`](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/if_nametoindex.3.html) and it's using _convert_ instead of _resolve_. I've changed the wording to match the man page a bit better.